scriptworker/cot/verify.py

```diff
@@ -803,36 +918,17 @@ def verify_cot_signatures(chain):
 
     """
     for link in chain.links:
-        path = link.get_artifact_full_path('public/chainOfTrust.json.asc')
-        gpg_home = os.path.join(chain.context.config['base_gpg_home_dir'], link.worker_impl)
-        gpg = GPG(chain.context, gpg_home=gpg_home)
-        log.debug("Verifying the {} {} chain of trust signature against {}".format(
-            link.name, link.task_id, gpg_home
-        ))
-        try:
-            with open(path, "r") as fh:
-                contents = fh.read()
-        except OSError as exc:
-            if is_task_required_by_any_mandatory_artifact(chain, link.task_id):
-                raise CoTError("Can't read mandatory {}: {}!".format(path, str(exc)))
-            else:
-                log.warning("Could not read optional {}. Continuing. Error gotten: {}!".format(path, str(exc)))
-                continue
+        unsigned_path = link.get_artifact_full_path('chain-of-trust.json')
+        ed25519_signature_path = link.get_artifact_full_path('chain-of-trust.json.sig')
```

Comment:
Do we want to somehow encode the format of the signature? Either in the filename, or by having the signature file be structured somehow? @alex, is there any standard practice around this, for detached signatures?

Reply:
It looks like the signature file has the body in it.
That raises the question of whether we publish a single file (the signature+body) or two (unsigned, and signature+body). I do foresee some automation using the unsigned json directly. If we're able to roll out a tool that can verify the signature and return the json (or ignore the signature and return the json), we could publish one file.

Reply:
Ah, the signature file has the body in it if I use `signing_key.sign(binary_contents)`, and is a detached signature if I use `signing_key.sign(binary_contents).signature`.

I'm open to standard practices.